doc/nrf/security/ap_protect.rst

```diff
@@ -197,16 +209,80 @@ This is the default setting in the |NCS|.
 
 You can start debugging the firmware without additional steps needed.
 
-Once you are done debugging, run the following command to enable the access port protection:
+.. _app_approtect_uicr_approtect:
+
+Enabling AP-Protect by locking the UICR.APPROTECT
+=================================================
+
+For the devices in production, you are highly recommended to lock the UICR.APPROTECT register to prevent unauthorized access to the device. The access port protection configured in this way cannot be disabled without erasing the flash.
+
+.. note::
+    This is the only mechanism supported for the nRF52 series and nRF9160 devices that do not have firmware support for AP-Protect.
+
+To lock the UICR.APPROTECT register, follow these steps:
 
 .. code-block:: console
 
    nrfjprog --rbp ALL
 
-This command enables the AP-Protect and resets the device.
+This command enables the AP-Protect (and Secure AP-Protect) and resets the device.
+
+Secure AP-Protect
+=================
+
+With Trusted Firmware-M (TF-M) we have a Secure Processing Environment (SPE) that is isolated from the Non-Secure Processing Environment (NSPE). TF-M is available for the nRF53 and nRF91 Series devices.
+While AP-Protect blocks access to all CPU registers and memories, Secure AP-Protect limits access to the CPU to only non-secure accesses.
+This allows debugging of the NSPE, while SPE debugging is blocked.
+
+The following Kconfig settings are available for nRF91x1 and nRF53 Series devices:
+
+Enabling Secure AP-Protect with :kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_LOCK`
+==================================================================================
```

Comment:
Consider making it a subsection of Secure AP-Protect. And the following three sections as well.